c7n/resources/sqs.py

```diff
@@ -70,3 +72,84 @@ def _augment(r):
 @SQS.filter_registry.register('cross-account')
 class SQSCrossAccount(CrossAccountAccessFilter):
     permissions = ('sqs:GetQueueAttributes',)
+
+
+@SQS.action_registry.register('delete')
+class DeleteSqsQueue(BaseAction):
+    """Action to delete a SQS queue
+
+    To prevent unwanted deletion of SQS queues, it is recommended
+    to include a filter
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: sqs-delete
+                resource: sqs
+                filters:
+                  - KmsMasterKeyId: absent
+                actions:
+                  - type: delete
+    """
+
+    schema = type_schema('delete')
+    permissions = ('sqs:DeleteQueue',)
+
+    def process(self, queues):
+        with self.executor_factory(max_workers=2) as w:
+            list(w.map(self.process_queue, queues))
+
+    def process_queue(self, queue):
+        client = local_session(self.manager.session_factory).client('sqs')
+        try:
+            client.delete_queue(QueueUrl=queue['QueueUrl'])
+        except ClientError as e:
+            self.log.exception(
+                "Exception deleting queue:\n %s" % e)
+
+
+@SQS.action_registry.register('set-encryption')
+class SetEncryption(BaseAction):
+    """Action to set encryption key on SQS queue
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: sqs-set-encrypt
+                resource: sqs
+                filters:
+                  - KmsMasterKeyId: absent
+                actions:
+                  - type: set_encryption
+                    key: "<alias of kms key>"
+    """
+    schema = type_schema(
+        'set-encryption',
+        key={'type': 'string'},required=('key',))
+
+    permissions = ('sqs:SetQueueAttributes','kms:DescribeKey',)
+
+    def process(self, queues):
+        with self.executor_factory(max_workers=2) as w:
+            list(w.map(self.process_queue, queues))
+
+    def process_queue(self, queue):
+        # get KeyId
+        key = "alias/" + self.data.get('key')
+        key_id = local_session(self.manager.session_factory).client(
```

Comment:
ie. the number of kms keys << number of queues, so doing queries per queue is less then efficient vs getting the vocab of keys for the set.